### PR TITLE
Add width and letter spacing configurability to Heading

### PR DESCRIPTION
**Pull request by rezrah**
## Summary

Part of https://github.com/github/galaxy.github.com/issues/118

Adds new `width` prop to `Heading` to allow granular control over variable font capabilities.

Widths based on presets defined in the [Mona Sans style table](https://github.com/github/mona-sans). 

Also added `letterSpacing` to allow granular adjustment over `letter-spacing` styles, which is necessary when updating the `width`

## Steps to test:

1. Check out the [docs page](https://primer-11843cfdec-26139705.drafts.github.io/components/Heading#width)
1. Check out the new stories
1. Review proposed API 

## Supporting resources (related issues, external links, etc):

- [Mona Sans](https://github.com/github/mona-sans)
- [Galaxy issue](https://github.com/github/galaxy.github.com/issues/118)

## Contributor checklist:

- [x] All new and existing CI checks pass
- [x] Tests prove that the feature works and covers both happy and unhappy paths
- [x] Any drop in coverage, breaking changes or regressions have been documented above
- [x] New visual snapshots have been generated / updated for any UI changes
- [x] All developer debugging and non-functional logging has been removed
- [x] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change

## Screenshots:
width
letter-spacing

### CODE REVIEW

File: packages/react/src/Heading/Heading.module.css [hunk 1]
```diff
@@ -144,3 +144,97 @@
...
   }
 }
+
+/**
+ * Heading widths
+ */
+
+.Heading--width-condensed,
+.Heading-narrow--width-condensed {
+  font-variation-settings: 'wdth' 75;
```
Comment (tobiasahlin) [inline packages/react/src/Heading/Heading.module.css:154]: `font-stretch` is also supported as an attribute, and accomplishes the same thing 👀 maybe move to that to allow `font-variation-settings` to be used for other settings (and reduce risk of conflicting utilities)? The value just need to be set in `%` ✨

File: apps/docs/content/components/Heading.mdx [hunk 2]
```diff
@@ -104,15 +107,85 @@ Use the `weight` prop to adjust the default visual appearance if required.
...
+
+```jsx live
+<>
+  <Heading as="h3" width="condensed">
+    condensed
+  </Heading>
+  <Heading as="h3" width="normal">
+    normal
+  </Heading>
+  <Heading as="h3" width="wide">
```
Comment (tobiasahlin) [inline apps/docs/content/components/Heading.mdx:124]: Am I misunderstanding the consts, or should this be `width="expanded"`? 🤔

**Review by tobiasahlin — COMMENTED** [review]
One comment and a question! Another thought: given that the CSS attribute is `font-stretch`, and we have a lot of things that have a `width`, it could make sense to move to and reserve `stretch` for setting the font width, to reduce confusion with other widths (e.g. viewport) 👀

File: packages/react/src/Heading/Heading.module.css (hunk 1 above)
Comment (rezrah) [reply packages/react/src/Heading/Heading.module.css:154]: Thanks @tobiasahlin! I've switched to `font-stretch` 👍

File: apps/docs/content/components/Heading.mdx (hunk 2 above)
Comment (rezrah) [reply apps/docs/content/components/Heading.mdx:124]: 🤦 Yes, thanks for spotting that. Docs look much better now. 

Screenshot 2023-02-14 at 17 16 34

**Comment by josepmartins** [comment]
> We have a lot of things that have a width, it could make sense to move to and reserve stretch for setting the font width

Agree with @tobiasahlin to use `stretch` instead of `width` to avoid confusion.

File: apps/docs/content/components/Heading.mdx [hunk 3]
```diff
@@ -104,15 +107,85 @@ Use the `weight` prop to adjust the default visual appearance if required.
...
+
+### Letter spacing
+
+The recommended character letter spacing is automatically applied to Headings based on their position on the type scale.
+
+Use the `letterSpacing` prop to override the default settings if required.
+
+```jsx live
+<>
+  <Heading as="h3" letterSpacing="condensed">
```
Comment (josepmartins) [inline apps/docs/content/components/Heading.mdx:152]:
```suggestion
  <Heading as="h3" letterSpacing="tight">
```

Same here. This should be changed, unless we stick with `condensed`

File: apps/docs/content/components/Heading.mdx [hunk 4]
```diff
@@ -104,15 +107,85 @@ Use the `weight` prop to adjust the default visual appearance if required.
...
+
+  {/* Responsive */}
+  <br />
+  <br />
+  <Heading
+    as="h4"
+    letterSpacing={{
+      narrow: 'condensed',
+      regular: 'normal',
+      wide: 'none',
```
Comment (josepmartins) [inline apps/docs/content/components/Heading.mdx:170]:
```suggestion
    letterSpacing={{
      narrow: 'tight',
      regular: 'normal',
      wide: 'none',
```

Depending on what we decide

File: packages/react/src/Heading/Heading.tsx [hunk 5]
```diff
@@ -7,9 +7,26 @@ export const HeadingSizes = ['1', '2', '3', '4', '5', '6'] as const
 export const HeadingTags = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6'] as const
 
 export const HeadingWeights = ['heavy', 'extrabold', 'bold', 'semibold', 'medium', 'normal', 'light'] as const
+export const HeadingWidths = ['condensed', 'normal', 'expanded'] as const
+export const HeadingLetterSpacing = ['tight', 'normal', 'none'] as const
```
[the remark this answers is not in the document]
Comment (rezrah) [reply packages/react/src/Heading/Heading.tsx]: I'm inclined to agree. 'tight' and 'condensed' are semantically interchangeable so happy to go with `condensed` for now. Have changed for now, but shout if anyone has other thoughts on. it.

File: .changeset/polite-bobcats-type.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+</Heading>
+<Heading as="h3" width="wide">
+  wide
+</Heading>
+```
+
+**Letter spacing**
+
+```jsx
+<Heading as="h3" letterSpacing="condensed">
```
[the remark this answers is not in the document]
Comment (rezrah) [reply .changeset/polite-bobcats-type.md:24]: Has been changed to condensed.

File: apps/docs/content/components/Heading.mdx (hunk 3 above)
Comment (rezrah) [reply apps/docs/content/components/Heading.mdx:152]: n/a as we switched to `condensed` 👍

**Review by tobiasahlin — APPROVED** [review]
Approving to not block ❤️ would love to get the move to the `stretch` namespace in, but also OK to do that in a follow-up PR (if you agree!)

**Comment by rezrah** [comment]
@tobiasahlin - the prop for `width` has been changed to `stretch`. You can see the latest changes in this commit https://github.com/primer/brand/pull/186/commits/fe89bf3bf9df59917c58b1ecdbf564821732dd2f. For consistency shall we change this everywhere else now? E.g. in the [Mona Sans repo under styles?](https://github.com/github/mona-sans)

**Review by tobiasahlin — APPROVED** [review]
Looks great! I'll get on changing width to stretch in the mona and hubot repos 🙏

**Pull request merged**